src/hotspot/share/c1/c1_GraphBuilder.cpp

```diff
@@ -2041,6 +2041,15 @@ void GraphBuilder::invoke(Bytecodes::Code code) {
         (code == Bytecodes::_invokevirtual && target->is_final_method()) ||
         code == Bytecodes::_invokedynamic) {
       ciMethod* inline_target = (cha_monomorphic_target != NULL) ? cha_monomorphic_target : target;
+
+      // We should not inline java.nio.Buffer.checkIndex if InlineNioCheckIndex is disabled
+      if (!InlineNIOCheckIndex &&
+          holder->is_instance_klass() &&
+          holder->as_instance_klass() == Compilation::current()->env()->nio_Buffer_klass() &&
+          inline_target->name() == ciSymbols::checkIndex_name()) {
+        return;
+      }
+
```

Comment:
Do we need to keep this flag?

Reply:
> Do we need to keep this flag?

Exactly, the flag should be removed.

Thanks,
Yang